software/sys_controller/ossc/av_controller.c

```diff
@@ -551,8 +552,11 @@ int load_profile() {
     retval = read_userdata(profile_sel);
     if (retval == 0) {
         // Change the input if the new profile demands a different one.
-        if (tc.link_av != AV_LAST && tc.link_av != cm.avinput)
+        // Also prevent the change of input from inducing a profile load.
+        if (tc.link_av != AV_LAST && tc.link_av != cm.avinput) {
             target_input = tc.link_av;
+            omit_profile_load = 1;
```

Comment:
Instead of omit_profile_load I would do
input_profiles[profile_link ? target_input : AV_TESTPAT] = profile_sel;
This would also avoid to reload another profile AND updates the input->profile link, what we have to do anyway if we want to be correctly, imo.

Reply:
If you want to save a tiny amount of code space, 
input_profiles[target_input] = profile_sel;
is also fine here, i.e. do not care about (profile_link == 0) as in this case input_profiles[AV_TESTPAT] equals profile_sel anyway.

Reply:
You're right that we do need to do that for correctness, and I am also happy to get rid of omit_profile_load. I'll do that change now.

Reply:
Done.